lib/test/unit/collector/load.rb

```diff
@@ -39,7 +39,7 @@ def collect(*froms)
             froms = @default_test_paths if froms.empty?
             froms = ["."] if froms.empty?
             test_suites = []
-            already_gathered = find_test_cases
+            already_gathered = find_test_cases.each_with_object({}) {|test_case, hash| hash[test_case] = true}
```

Comment:
Does this work? I want to update `already_gathered` only in one place (`find_test_cases` in this case) if we can. It's for maintainability.

```suggestion
            already_gathered = {}
            find_test_cases(already_gathered)
```

Reply:
It works fine. 😃 
I picked suggestions, squashed commits, and force pushed.